### PR TITLE
Counting backend

### PR DESCRIPTION
**Pull request by UwUAroze**
mriaow ;3

### CODE REVIEW

File: src/events/handlers/CountingHandler.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+export class CountingHandler implements IEvent {
+  public event: keyof ClientEvents = Events.MessageCreate;
+  public once = false;
+
+  constructor(private client: SnugglesClient) {}
+
+  async handle(message: OmitPartialGroupDMChannel<Message<boolean>>) {
+    const countData = await GuildCountingService.getCountingDataForChannel(message.channelId);
+    if (!countData) return;
+    logger.debug(`message: ${message.content} | count: ${countData.count}`)
```
Comment (lilysoftpaws) [inline src/events/handlers/CountingHandler.ts]: we can't have debug code in production...

File: src/events/handlers/CountingHandler.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      return
+    }
+
+    if (currentCount != 0 && !countData.allowConsecutiveCounts && lastCounter === message.author.id) {
+      countData.count = 0
+      await GuildCountingService.updateCountData(countData)
+      await message.reply(this.parseCountingMessage(countData.consecutiveCountingFailMessage, message.author, currentCount))
+      await message.react(this.getReaction(countData, isCorrect))
+      return
+    }
```
Comment (lilysoftpaws) [inline src/events/handlers/CountingHandler.ts:68]: the above two if-statements have very similar functions...

File: src/events/handlers/CountingHandler.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+    countData.count++
+    await GuildCountingService.updateCountData(countData)
+    await message.react(this.getReaction(countData, isCorrect))
+  }
+
+  private getReaction(countData: GuildCounting, isCorrect: boolean): GuildEmoji|string {
+    let emoji: GuildEmoji|string|undefined = undefined
```
Comment (lilysoftpaws) [inline src/events/handlers/CountingHandler.ts]: does your keyboard not have a spacebar

**Review by lilysoftpaws — CHANGES_REQUESTED** [review]

File: src/events/handlers/CountingHandler.ts (hunk 1 above)
Comment (UwUAroze) [reply src/events/handlers/CountingHandler.ts]: mY BAD

File: src/events/handlers/CountingHandler.ts (hunk 2 above)
Comment (UwUAroze) [reply src/events/handlers/CountingHandler.ts:68]: i made it _"expandable"_ ✨✨

File: src/events/handlers/CountingHandler.ts (hunk 3 above)
Comment (UwUAroze) [reply src/events/handlers/CountingHandler.ts]: dsfgljhdfghjl

File: src/events/handlers/CountingHandler.ts (hunk 2 above)
Comment (lilysoftpaws) [reply src/events/handlers/CountingHandler.ts:68]: it doesn't look very expandable

Comment (UwUAroze) [reply src/events/handlers/CountingHandler.ts:68]: well it is

**Pull request merged**